onnxruntime/core/optimizer/nchwc_transformer.cc

```diff
@@ -666,6 +666,125 @@ void NchwcTransformerImpl::TransformActivation(Node& node) {
+  if (bn_var_tensor_proto == nullptr) {
+    return;
+  }
+
+  auto bn_scale = onnxruntime::make_unique<Initializer>(*bn_scale_tensor_proto);
+  auto bn_B = onnxruntime::make_unique<Initializer>(*bn_B_tensor_proto);
+  auto bn_mean = onnxruntime::make_unique<Initializer>(*bn_mean_tensor_proto);
+  auto bn_var = onnxruntime::make_unique<Initializer>(*bn_var_tensor_proto);
+
+  // Calculate the scale and bias for the replacement convolution.
+  bn_var->add(epsilon);
+  bn_var->sqrt();
+  bn_scale->div(*bn_var);
+  bn_mean->mul(*bn_scale);
+  bn_B->sub(*bn_mean);
+
+  const size_t nchwc_block_size = MlasNchwcGetBlockSize();
+  const int64_t nchwc_channels = (channels + nchwc_block_size - 1) & ~(nchwc_block_size - 1);
+
+  std::vector<float> padded_buffer(nchwc_channels);
+
+  std::copy_n(bn_scale->data<float>(), channels, padded_buffer.data());
+
+  ONNX_NAMESPACE::TensorProto nchwc_conv_W_tensor_proto;
+  nchwc_conv_W_tensor_proto.set_data_type(ONNX_NAMESPACE::TensorProto_DataType_FLOAT);
+  nchwc_conv_W_tensor_proto.set_name(graph_.GenerateNodeArgName("bn_scale"));
+  nchwc_conv_W_tensor_proto.set_raw_data(padded_buffer.data(), nchwc_channels * sizeof(float));
+  nchwc_conv_W_tensor_proto.add_dims(nchwc_channels);
+  nchwc_conv_W_tensor_proto.add_dims(1);
+  nchwc_conv_W_tensor_proto.add_dims(1);
+  nchwc_conv_W_tensor_proto.add_dims(1);
+
+  graph_.AddInitializedTensor(nchwc_conv_W_tensor_proto);
+
+  auto* nchwc_conv_W_arg = &graph_.GetOrCreateNodeArg(nchwc_conv_W_tensor_proto.name(), nullptr);
+
+  std::copy_n(bn_B->data<float>(), channels, padded_buffer.data());
+
+  ONNX_NAMESPACE::TensorProto nchwc_conv_B_tensor_proto;
+  nchwc_conv_B_tensor_proto.set_data_type(ONNX_NAMESPACE::TensorProto_DataType_FLOAT);
+  nchwc_conv_B_tensor_proto.set_name(graph_.GenerateNodeArgName("bn_B"));
+  nchwc_conv_B_tensor_proto.set_raw_data(padded_buffer.data(), nchwc_channels * sizeof(float));
+  nchwc_conv_B_tensor_proto.add_dims(nchwc_channels);
+
+  graph_.AddInitializedTensor(nchwc_conv_B_tensor_proto);
+
+  auto* nchwc_conv_B_arg = &graph_.GetOrCreateNodeArg(nchwc_conv_B_tensor_proto.name(), nullptr);
+
+  // Create the replacement node.
+  std::string nchwc_node_name = graph_.GenerateNodeName(output_defs[0]->Name() + "_bn_nchwc");
+  Node& nchwc_node = graph_.AddNode(nchwc_node_name,
+                                    "Conv",
+                                    nchwc_node_name,
+                                    {nchwc_input->nchwc_arg_, nchwc_conv_W_arg, nchwc_conv_B_arg},
+                                    output_defs,
+                                    nullptr,
+                                    kMSNchwcDomain);
+  nchwc_node.SetExecutionProviderType(node.GetExecutionProviderType());
+  nchwc_node.AddAttribute("group", nchwc_channels);
+
+  nchwc_input->remaining_original_uses_--;
+
+  CreateNchwcArgument(node, nchwc_node, channels, nchwc_input->shape_);
+  removed_nodes_.push_front(node.Index());
+}
```

Comment:
FWIW: At some point it would be nice to handle moving edges when replacing nodes so that the expensive call to Graph::Resolve can eventually be avoided after running every transformer that modifies the graph. e.g. graph_utils::FinalizeNodeFusion does this sort of thing. I don't think there's currently a helper in graph_utils to handle this exact scenario of swapping a new Node with an existing one though.

Just a general note and not required for this PR as I think all the transformers here should be reviewed and updated this way at once if possible.